### PR TITLE
Remove driver#initializePachConn and auth#In2Out

### PR DESCRIPTION
**Pull request by msteffen**

### CODE REVIEW

File: src/server/pfs/server/driver.go [hunk 1]
```diff
@@ -90,11 +90,11 @@ type driver struct {
...
 	address      string
 	pachConnOnce sync.Once
-	onceErr      error
 	pachConn     *grpc.ClientConn
 
 	// pachClient is a cached Pachd client, that connects to Pachyderm's object
-	// store API and auth API
+	// store API and auth API. Instead of accessing it directly, functions should
+	// call getPachClient()
 	pachClient *client.APIClient
```
Comment (sjezewski) [inline src/server/pfs/server/driver.go]: Is there a convention we can supply here to further mark it private? Maybe name it `_pachClient` ?

File: src/server/pfs/server/driver.go [hunk 2]
```diff
@@ -158,33 +158,37 @@ func newLocalDriver(blockAddress string, etcdPrefix string) (*driver, error) {
...
-		d.pachConn, d.onceErr = grpc.Dial(d.address, client.PachDialOptions()...)
+		var err error
+		d.pachConn, err = grpc.Dial(d.address, client.PachDialOptions()...)
 		d.pachClient = &client.APIClient{
 			AuthAPIClient:   auth.NewAPIClient(d.pachConn),
 			ObjectAPIClient: pfs.NewObjectAPIClient(d.pachConn),
 		}
+		if err != nil {
+			panic(fmt.Sprintf("could not intiailize Pachyderm client in driver: %v", err))
+		}
```
Comment (sjezewski) [inline src/server/pfs/server/driver.go]: So what is the desired recourse? If on startup the driver cannot successfully dial we panic and force a restart? Is there a reason this isn't in a backoff (or at least backoff pending the network is retryable pattern)?

File: src/client/auth/auth.go [hunk 3]
```diff
@@ -47,22 +45,6 @@ func ParseScope(s string) (Scope, error) {
 	return Scope_NONE, fmt.Errorf("unrecognized scope: %s", s)
 }
 
-// In2Out converts an incoming context containing auth information into an
-// outgoing context containing auth information, stripping other keys (e.g.
-// for metrics) in the process. If the incoming context doesn't have any auth
-// information, then the returned context won't either.
-func In2Out(ctx context.Context) context.Context {
```
Comment (sjezewski) [inline src/client/auth/auth.go]: I'm missing something. Why is this no longer needed?

Comment (msteffen) [reply src/client/auth/auth.go]: It's no longer called :-). Basically, when you call `pachClient.WithCtx(ctx)`, it does its own transformation of the incoming ctx before copying it to the new client struct. All of Pachyderm's code now uses that abstraction instead

Comment (sjezewski) [reply src/client/auth/auth.go]: :+1:

**Pull request merged**